src/scripts/components/DropdownCheckbox.jsx

```diff
@@ -0,0 +1,268 @@
+
+    // Close dropdown
+    $dropdown.timeline.reverse();
+
+    this.setState({ isOpen: false });
+
+    if (this.props.onClick) {
+      this.props.onClick();
+    }
+  }
+
+  updateActiveKey = (index) => {
+    if (this.props.hideActive) {
+      return;
+    }
+
+    this.setState({
+      activeKey: index,
+    });
+  }
+
+  render() {
+    const { block, className, disabled, disableScroll, hideCaret, label, icon, lockLabel, position, reset, size, title, type, wide, onStart, onComplete, onReverseStart, onReverseComplete, showOverflow, wrapperClassName, showAssociatedLabel, checkboxClassName } = this.props;
+    const { activeKey, hasFilter } = this.state;
+
+    const dropdownClasses = cx('dropdown', {
+      'dropdown--block': block,
+    }, wrapperClassName);
+
+    const dropdownToggleClasses = cx('dropdown__toggle', className);
+
+    const dropdownMenuClasses = cx('dropdown__menu', {
+      'dropdown__menu--right': position === 'right',
+      'dropdown__menu--center': position === 'center',
+      'dropdown__menu--top': position === 'top',
+      'dropdown__menu--top dropdown__menu--right': position === 'top-right',
+      'dropdown__menu--top dropdown__menu--center': position === 'top-center',
+      'dropdown__menu--wide': wide,
+    });
+
+    let selectedLabel = null;
+    let selectedIcon = null;
+
+    if (activeKey && !lockLabel) {
+      React.Children.forEach(this.props.children, (child) => {
+        if (!child) return;
+        if (child.type === DropdownMenuItem) {
+          if (child.props.id === activeKey) {
+            selectedLabel = child.props.label;
+            selectedIcon = child.props.icon || child.props.labelIcon;
+          }
+        } else if (child.type === DropdownFilter) {
+          React.Children.forEach(child.props.children, (filterChild) => {
+            if (filterChild.type === DropdownMenuItem) {
+              if (filterChild.props.id === activeKey) {
+                selectedLabel = filterChild.props.label;
+                selectedIcon = filterChild.props.icon || child.props.labelIcon;
+              }
+            }
+          });
+        }
+      });
+    }
+
+    const enableClickOutside = this.props.manualClose ? false : this.state.isOpen;
+    const showLabel = () => {
+      const { labelValueAssociated } = this.props;
+      if (showAssociatedLabel) {
+        return <span className="dropdown__toggle__text">{labelValueAssociated}</span>;
+      }
+      if (selectedLabel || label) {
+        if (showOverflow) {
+          return selectedLabel || label;
+        }
+
+        return <span className="dropdown__toggle__text">{selectedLabel || label}</span>;
+      }
+      return false;
+    };
+    const showCheckbox = () => {
+      const { isCheckbox } = this.state;
+      if (isCheckbox) {
+        return <Checkbox name="test" label=" " isChecked={this.state.checked} onChange={this.props.onChange} className={checkboxClassName} />;
```

Comment:
Please use destructuring here also for `this.state.checked` and `this.props.onChange`